### ISSUE TITLE
Allow candidate to complete test at home

### ISSUE THREAD

**Issue by johnmccollum** [arnoldclark/front-end-interview#13]
# The Problem

I was recently a wee bit surprised to see a poll (n=136) of developers where two-thirds preferred take-home tasks to live coding. This has been reflected anecdotally in some conversations that I've had with people.

<img width="485" alt="Screenshot 2022-11-10 at 14 19 15" src="https://user-images.githubusercontent.com/48916396/201115962-8042cefe-796d-4186-bffe-ac02cd034a2a.png">

This is a bit of an issue because we lean heavily into the live coding side. (**n.b.** - this is just one data point, and I suspect "live coding" in other interviews is much more stressful than the way we do it here - "solve this problem" as opposed to the mob programming approach we take.)

Do we need to revisit our approach? Is there any downside to offering this as an option alongside what we're currently doing?

## Option 1

If we feel we would like to offer the option of take-home task, one way to do it would be to allow the candidate to complete the "Garage" task at home. So a combination of the task from the mob programming side, with the approach of "show and tell".

We would probably need to think about the wording of the task, and there might be bits we could rework to make the project easier to run, but is this something we should consider?

**Comment by futuresocks** [comment]
Yeah, this wouldn't immediately make me abandon the approach we've cultivated, which by and large I think is pretty sound. That said, our approach won't be for everyone, so by allowing a candidate the option of doing it at home we'd be providing more flexibility and choice for candidates. That can only be a good thing, I reckon.

Initially I had pause over if an option like this would mean we'd lose anything that we'd normally glean from a live coding exercise. However, given that we already provide candidates the chance to show and tell around something they've worked on, all we'd be doing by providing this option would be introducing a... controlled? show and tell? Where the interviewers would be more au fait with what's being presented. No drawbacks imho.

**Comment by PrincessSarahB** [comment]
I agree, I think this is a good idea as everyone is different. Person 1 might be super up for a wee mob programming task and perform really well in that environment, whereas person 2 might be way more nervous about doing that in an interview situation and prefer to show and tell a task we've given them. Giving them the option means we're less likely to lose out on good candidates who maybe wouldn't have performed so well doing the mobbing task because of nerves etc. and also removes the whole having to decide which of their own work to show us and worrying if it is the right kind of thing.

**Comment by AMalloch** [comment]
I would lean more towards the one that suits the company's goals rather than the preference if that makes sense. But that said if all bases are covered then offering more options to people can only be a good thing.

**Comment by khaledkz** [comment]
I think we should give them both options, but the technical task should be different for each option, for mob programming we can give the candidate the chance to represent his skills, like problem solving and communication and ability to think under pressure but in take home task they should create a simple project to represent they coding knowledge and how they approach the tasks.

I think junior developers do mind to spend more time and do good home task but senior developers would prefer live technical tasks and if they do home task they will spend less time in it and that could make some confusing about the candidate.
